url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/geom/Segment3D.java

```diff
@@ -0,0 +1,1077 @@
+	 * 
+	 * @param point
+	 *            Point3D to test
+	 * @return boolean - true if contained, false otherwise.
+	 * @throws ArithmeticException
+	 *             - if norm equals zero in epsilon range. This exception
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 * @see #containsOnLine(Point3D, ScalarOperator)
+	 */
+	public boolean contains(Point3D point, ScalarOperator sop) {
+		if (new Line3D(this, sop).contains(point, sop))
+			if (this.containsOnLine(point, sop))
+				return true;
+
+		return false;
+	}
+
+	/**
+	 * Tests whether a segment is contained on this.
+	 * 
+	 * @param seg
+	 *            Segment3D for test
+	 * @param sop
+	 *            ScalarOperator
+	 * @return boolean - true if contained, false otherwise.
+	 * @throws ArithmeticException
+	 *             - if norm equals zero in epsilon range. This exception
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 */
+	public boolean contains(Segment3D seg, ScalarOperator sop) {
+		Point3D[] points = seg.getPoints();
+		if (this.contains(points[0], sop) && this.contains(points[1], sop))
+			return true;
+		else
+			return false;
+	}
+
+	/**
+	 * Computes the intersection between this and the given Line3D.<br>
+	 * Returns this if the Line3D and this are parallel and their distance is
+	 * equal 0.<br>
+	 * Returns null if they are parallel and their distance is ! = 0 or they are
+	 * skew, or they not intersect.<br>
+	 * Returns a Point3D object if they intersect.
+	 * 
+	 * @param line
+	 *            Line3D for intersection
+	 * @param sop
+	 *            ScalarOperator
+	 * @return SimpleGeoObj - result SimpleGeoObj object (null, Segment3D,
+	 *         Point3D).
+	 * @throws IllegalStateException
+	 *             - if the intersectsInt(Line3D line, ScalarOperator sop)
+	 *             method (which computes the intersection of two lines) of the
+	 *             class Line3D returns a value that is not -2, -1, 0 or 1.
+	 * @throws ArithmeticException
+	 *             - if norm equals zero in epsilon range. This exception
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 */
+	public SimpleGeoObj intersection(Line3D line, ScalarOperator sop) {
+		return line.intersection(this, sop);
+	}
+
+	/**
+	 * Computes the intersection between this and the given Segment3D.<br>
+	 * Assumes that the given segment lies on line of this (and vice versa)
+	 * Returns <code>null</code> if they do not intersect.<br>
+	 * Returns a Point3D object if they touch each other in end points.<br>
+	 * Returns a Segment3D if they intersect strict.
+	 * 
+	 * @param seg
+	 *            Segment3D for intersection
+	 * @return SimpleGeoObj - result SimpleGeoObj object (null, Segment3D,
+	 *         Point3D).
+	 */
+	public SimpleGeoObj intersectionOnLine(Segment3D seg, ScalarOperator sop) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.